dask_jobqueue/core.py

```diff
@@ -226,6 +233,26 @@ def __init__(
             warnings.warn(warn, FutureWarning)
             if not worker_extra_args:
                 worker_extra_args = extra
+
+        if job_extra is None:
+            job_extra = dask.config.get("jobqueue.%s.job-extra" % self.config_name, {})
+        if job_extra_directives is None:
+            job_extra_directives = dask.config.get(
+                "jobqueue.%s.job-extra-directives" % self.config_name, {}
```

Comment:
We should probably put None as the default, or no default, for both job_extra and job_extra_directives?

Reply:
For other options (including the old `job_extra`), the default (in the code and *at least* in the yaml-config) seems to be an empty list or dict such that you always have something iterable to do things like
```python
header_lines.extend(["#SBATCH %s" % arg for arg in self.job_extra_directives])
```
without checking if it is empty/None first. (This was at least my interpretation of this general pattern here.)

You are probably right, at this place one could drop the default, but I would certainly keep it in the config, then. Having an empty dict here should be fine, though, for all cases -- up to you to decide.

Reply:
I would at least replace the dict by a list, as most implementations use a list here. Would that work for HTCondor?

Reply:
I chose dict, because I considered that more general: you can iterate over a dict, i.e., over the values, but you can't use `dict.update()` with a list as argument.
But you have a point here and in the case of an empty list, it also works for HTCondor. I just pushed the change.